pkg/helpers/testing/assertion.go

```diff
@@ -0,0 +1,174 @@
+package testing
+
+import (
+	"reflect"
+	"testing"
+
+	clusterv1 "github.com/open-cluster-management/api/cluster/v1"
+
+	authorizationv1 "k8s.io/api/authorization/v1"
+	certv1beta1 "k8s.io/api/certificates/v1beta1"
+	coordinationv1 "k8s.io/api/coordination/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	clienttesting "k8s.io/client-go/testing"
+	"k8s.io/utils/diff"
+)
+
+func AssertError(t *testing.T, actual error, expectedErr string) {
+	if len(expectedErr) > 0 && actual == nil {
+		t.Errorf("expected %q error", expectedErr)
+		return
+	}
+	if len(expectedErr) > 0 && actual != nil && actual.Error() != expectedErr {
+		t.Errorf("expected %q error, but got %q", expectedErr, actual.Error())
+		return
+	}
+	if len(expectedErr) == 0 && actual != nil {
+		t.Errorf("unexpected err: %v", actual)
+		return
+	}
+}
+
+func AssertActions(t *testing.T, actualActions []clienttesting.Action, expectedActions ...string) {
```

Comment:
I recommend name change and doc. `expectedVerbs`?

Reply:
renamed and add the doc for each  assert method